File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int currentGain = 0;
        for (int i = 1; i < prices.size(); i++) {
            auto gain = prices[i] - prices[i-1];
            if ( gain > 0 ) currentGain += gain;
            else {
                res += currentGain;
                currentGain = 0;
            }
        }
        res += currentGain;
        return res;
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut res: i32 = 0;
        for index in (1..prices.len()) {
            let gain = prices[index] - prices[index - 1];
            res += 0.max(gain);
        }
        res
    }
}